From 8d8be57f0e29a9de0444b14a17c87e85e55dc2f0 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 19 Jan 2022 07:59:17 +0200
Subject: [PATCH] [SW-58071] common: add weak and unused macro

This patch defines macros for weak and unused attributes and uses
it throughout code, it helps improve code readability.

Change-Id: Iba4c240b4661a5d21145a4abc2b75bc736deb22c
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index a173502..e0dd757 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -27,7 +27,7 @@
 #define ARCBOOT_BRINGUP_ERROR		-1
 #define ARCBOOT_BRINGUP_OK		1
 
-__attribute__((weak)) void arcboot_rom_status_update(enum arcboot_rom_progress progress)
+__weak void arcboot_rom_status_update(enum arcboot_rom_progress progress)
 {
 	uint32_t reg;
 
@@ -108,9 +108,9 @@
 	}
 }
 
-__attribute__((weak)) void update_warm_reset_config(void){}
+__weak void update_warm_reset_config(void){}
 
-__attribute__((weak)) void security_disable_interfaces(void)
+__weak void security_disable_interfaces(void)
 {
 	/* by default disable jtag I/F as soon as possible */
 	security_disable_jtag_if();
@@ -345,7 +345,7 @@
 	return ARCBOOT_BRINGUP_OK;
 }
 
-__attribute__((weak)) void arcboot_rom_handle_tpm(void) {}
+__weak void arcboot_rom_handle_tpm(void) {}
 
 static void arcboot_rom_bringup_pcie(void)
 {
diff --git a/common/efuse.c b/common/efuse.c
index 5c5eb59..e2aa755 100644
--- a/common/efuse.c
+++ b/common/efuse.c
@@ -53,7 +53,7 @@
 	return 0;
 }
 
-__attribute__((weak)) int fuse_read(uint32_t word, uint32_t *val)
+__weak int fuse_read(uint32_t word, uint32_t *val)
 {
 	int offset;
 
diff --git a/common/pcie.c b/common/pcie.c
index 47ce067..f5b50e9 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -169,7 +169,7 @@
 		pcie_config();
 }
 
-__attribute__((weak))void execute_pcie_recovery(void)
+__weak void execute_pcie_recovery(void)
 {
 	pcie_recovery();
 }
diff --git a/common/rsa/rsa-verify.c b/common/rsa/rsa-verify.c
index dc352be..0ad3468 100644
--- a/common/rsa/rsa-verify.c
+++ b/common/rsa/rsa-verify.c
@@ -7,6 +7,7 @@
 #include <rsa.h>
 #include <sha512.h>
 #include <string.h>
+#include <arcboot.h>
 
 /*
  * This array is declared as global, to minimise
@@ -49,7 +50,7 @@
 
 int padding_pkcs_15_verify(uint8_t *msg, int msg_len,
 			   const uint8_t *hash,
-		__attribute__ ((unused))int hash_len)
+		__unused int hash_len)
 {
 	int ret, pad_len = msg_len - SHA384_SUM_LEN;
 
@@ -156,7 +157,7 @@
  *  ------------------------
  */
 int rsa_verify_with_pkey(uint8_t *key,
-		 	__attribute__ ((unused))uint32_t key_len,
+			__unused uint32_t key_len,
 			 const void *hash, uint8_t *sig, uint32_t sig_len)
 {
 	struct key_prop prop;
diff --git a/common/security.c b/common/security.c
index 2fca9a5..1ccfcaa 100644
--- a/common/security.c
+++ b/common/security.c
@@ -116,7 +116,7 @@
 	return -1;
 }
 
-__attribute__((weak)) int security_get_key_hash(uint32_t *hash, uint8_t customer)
+__weak int security_get_key_hash(uint32_t *hash, uint8_t customer)
 {
 	uint32_t valid_key_base;
 	int i;
diff --git a/common/sha512.c b/common/sha512.c
index 2276bfb..bef1d32 100644
--- a/common/sha512.c
+++ b/common/sha512.c
@@ -12,6 +12,7 @@
 
 #include <string.h>
 #include <sha512.h>
+#include <arcboot.h>
 
 const uint8_t sha384_der_prefix[SHA384_DER_LEN] = {
 	0x30, 0x41, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86,
@@ -282,7 +283,7 @@
  */
 void sha384_csum(const unsigned char *input, unsigned int ilen,
 		unsigned char *output,
-		__attribute__ ((unused))unsigned int chunk_sz)
+		__unused unsigned int chunk_sz)
 {
 	sha512_context ctx;
 
diff --git a/common/sha512.h b/common/sha512.h
index e5a5a55..3f54898 100644
--- a/common/sha512.h
+++ b/common/sha512.h
@@ -2,6 +2,7 @@
 #define _SHA512_H
 
 #include <stdint.h>
+#include <arcboot.h>
 
 #define SHA384_SUM_LEN          48
 #define SHA384_DER_LEN          19
@@ -28,7 +29,7 @@
 
 void sha384_csum(const unsigned char *input, unsigned int ilen,
 		unsigned char *output,
-		__attribute__ ((unused))unsigned int chunk_sz);
+		__unused unsigned int chunk_sz);
 
 
 #endif /* _SHA512_H */
diff --git a/include/arcboot.h b/include/arcboot.h
index c3a25c2..d980b12 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -14,6 +14,14 @@
 #include "regmap.h"
 #include "regmask.h"
 
+#ifndef __weak
+#define __weak		__attribute__((weak))
+#endif
+
+#ifndef __unused
+#define __unused	__attribute__((__unused__))
+#endif
+
 #define BIT(nr)					(1 << (nr))
 #define ARRAY_SIZE(x)				(sizeof(x) / sizeof((x)[0]))
 
